elki/src/main/java/de/lmu/ifi/dbs/elki/algorithm/clustering/kmeans/initialization/PredefinedInitialMeans.java

```diff
@@ -94,9 +94,7 @@ public void setInitialClusters(List<? extends Cluster<? extends MeanModel>> init
    */
   public void setInitialMeans(double[][] initialMeans) {
     double[][] vecs = new double[initialMeans.length][];
-    for(int i = 0; i < initialMeans.length; ++i) {
-      vecs[i] = initialMeans[i]; // TODO: clone?
-    }
+    System.arraycopy(initialMeans, 0, vecs, 0, initialMeans.length);
```

Comment:
Yes, the TODO is "do we need to clone here, or is it always safe to reuse". I'm not sure if the initial-cluster-centers API has been well enough specified. I believe most algorithms assume they are allowed to modify the double[] arrays at will; and most initializations will create a fresh array, too. But there could be some unintended side effects.